backend/client.py

```diff
@@ -0,0 +1,79 @@
+import logging
+from diart import OnlineSpeakerDiarization
+from audio_source import StreamingSocketAudioSource
+from config import DIARIZATION_PIPELINE_CONFIG, LANGUAGE_MAPPING
+from whisper_transcriber import WhisperTranscriber
+import asyncio
+from queue import Queue
+
+
+class Client:
+
+    def __init__(self, sid, socket, transcription_thread, chunk_receiving_thread, config):
+        self.diarization_pipeline = OnlineSpeakerDiarization(DIARIZATION_PIPELINE_CONFIG)
+        self.pipeline_config = DIARIZATION_PIPELINE_CONFIG
+        self.transcriber = None
+        self.config = config
+        self.source = StreamingSocketAudioSource(sid)
+        self.socket = socket
+        self.transcription_thread = transcription_thread
+        self.chunk_receiving_thread = chunk_receiving_thread
+        self.audio_chunks = Queue()
+
+        self.initialize_transcriber()
+
+    def initialize_transcriber(self):
+        model = self.config.get("model", "small")
```

Comment:
This would be the implementation:

1. Definition of the enum in the config file

```python
class WhisperModelSize(Enum):
    TINY = 'tiny'
    SMALL = 'small'
    MEDIUM = 'medium'
    LARGE_V1 = 'large-v1'
    LARGE_V2 = 'large-v2'
```

2. Utility function in utils.py to format model size names received from the client to match the enum members' names

```
def format_whisper_model_name(model_name):
    """
    Format the model size name (tiny, small, large-v1, etc) to match the enum members in WhisperModelSize

    Replaces dashes with underscores and makes the string uppercase
    """
    return model_name.replace("-", "_").upper()
```

3. Retrieve the corresponding enum member when initializing the transcriber, default to small if doesn't exist

```python
# Format the name received from the client to match the enum members
whisper_model_name = format_whisper_model_name(self.config.get("model", "small"))
try:
    whisper_model = WhisperModelSize(whisper_model_name)
except ValueError:
    logging.warning("Invalid model size, defaulting to small")
    whisper_model = WhisperModelSize.SMALL
```

4. Pass the given model's value as an argument to WhisperTranscriber's init

```python
self.transcriber = WhisperTranscriber(model_name=whisper_model.value, language_code=language_code)
```

Reply:
Looks good!